### PR TITLE
chatroom: State._process is an async generator

### PR DESCRIPTION
**Pull request by masenf**
use `async for` construct when iterating through state updates generated from processing events.

use form and on_submit to get easy enter-button send functionality

fixes #124

### CODE REVIEW

**Comment by milochen0418** [comment]
It works 
Screenshot 2023-06-29 at 12 31 06 PM

**Review by milochen0418 — APPROVED** [review]
Screenshot 2023-06-29 at 12 31 06 PM
It works ~

**Review by milochen0418 — APPROVED** [review]

File: chatroom/requirements.txt
Comment (Lendemor) [inline chatroom/requirements.txt:1]: should pass it to reflex>=0.2.0

File: chatroom/chatroom/chatroom.py
Comment (Lendemor) [inline chatroom/chatroom/chatroom.py:1]: let's pass it on reflex too and use `rx.`

**Review by Lendemor — CHANGES_REQUESTED** [review]

File: chatroom/chatroom/chatroom.py
Comment (masenf) [reply chatroom/chatroom/chatroom.py:1]: done

**Review by milochen0418 — APPROVED** [review]

**Review by Lendemor — APPROVED** [review]

**Pull request merged**